### PR TITLE
feat(deployment): data_plane does not write data to etcd

### LINKED ISSUE #6735: feat: As a user, I want to start Apache APISIX with a etcd user (who only have read permission of etcd), so that we can improve the security and avoid changing the etcd in DP side
### Description

Hello,  I config the Apache APISIX with a user (who only has read permission of `/apisix/`), and when I start Apache APISIX.
I meet this problem: 
```
got malformed key-put message: "{"error":"etcdserver: permission denied","code":7,"message":"etcdserver: permission denied"}" from etcd "http://127.0.0.1:2379/v3/kv/put
```

The code is as follow:
https://github.com/apache/apisix/blob/55eb58ab85ac7e108de36271fd9cf29fe2fa3324/apisix/cli/etcd.lua#L310-L365

I think we can avoid adding the `init_dir` if the directory exists.

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes #6735

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by soulbird — DISMISSED** [review]

File: docs/en/latest/plugins/server-info.md [hunk 1]
```diff
@@ -41,6 +41,8 @@ The information reported by the Plugin is explained below:
 | version      | string  | Version of APISIX instance.                                                                                            |
 | hostname     | string  | Hostname of the machine/pod APISIX is deployed to.                                                                     |
 
+**NOTE：When the APISIX role is set to `data_plane` in deployment mode, unable to use the plugin**
```
Comment (tokers) [inline docs/en/latest/plugins/server-info.md:44]: Can we use the docusaurus admonitions?

```
:::warning

This plugin won't work if the APISIX role is `data_plane`.

:::
```

Comment (tzssangglass) [reply docs/en/latest/plugins/server-info.md]: update

File: docs/en/latest/plugins/server-info.md [hunk 2]
```diff
@@ -41,6 +41,12 @@ The information reported by the Plugin is explained below:
 | version      | string  | Version of APISIX instance.                                                                                            |
 | hostname     | string  | Hostname of the machine/pod APISIX is deployed to.                                                                     |
 
+:::warning
+
+This plugin won't work if the APISIX role is `data_plane`.
```
Comment (spacewander) [inline docs/en/latest/plugins/server-info.md:46]: We don't need a notice as there is always a report from CP or traditional role.

File: docs/en/latest/plugins/request-id.md [hunk 3]
```diff
@@ -47,6 +47,12 @@ The Plugin will not add a unique ID if the request already has a header with the
 | include_in_response | boolean | False    | true           |                                 | When set to `true`, adds the unique request ID in the response header. |
 | algorithm           | string  | False    | "uuid"         | ["uuid", "snowflake", "nanoid"] | Algorithm to use for generating the unique request ID.                 |
 
+:::warning
+
+The `snowflake` algorithm won't work if the APISIX role is `data_plane`.
```
Comment (spacewander) [inline docs/en/latest/plugins/request-id.md:52]:
```suggestion
When you need to use `snowflake` algorithm, make sure APISIX has the permission to write to the etcd.
```

File: docs/zh/latest/plugins/request-id.md [hunk 4]
```diff
@@ -33,6 +33,12 @@ title: request-id
 | include_in_response | boolean | 可选 | true          |                       | 是否需要在返回头中包含该唯一 ID |
 | algorithm           | string  | 可选 | "uuid"         | ["uuid", "snowflake", "nanoid"] | ID 生成算法 |
 
+:::warning
+
+当部署模式中设置 APISIX 角色为 `data_plane` 时，无法使用 `snowflake` 算法。
```
Comment (spacewander) [inline docs/zh/latest/plugins/request-id.md:38]:
```suggestion
当使用 `snowflake` 算法时，请确保 APISIX 有权限写入 etcd。
```

Comment (tzssangglass) [reply docs/zh/latest/plugins/request-id.md]: updated

File: docs/en/latest/plugins/request-id.md (hunk 3 above)
Comment (tzssangglass) [reply docs/en/latest/plugins/request-id.md]: updated

File: docs/en/latest/plugins/server-info.md (hunk 2 above)
Comment (tzssangglass) [reply docs/en/latest/plugins/server-info.md]: rm it

**Review by starsz — APPROVED** [review]

**Pull request merged**